core/src/main/java/org/radargun/stages/cache/background/LogChecker.java

```diff
@@ -110,32 +108,36 @@ public void run() {
                   LastOperation lastCheck = (LastOperation) last;
                   record = newRecord(record, lastCheck.getOperationId(), lastCheck.getSeed());
                }
+               checkIgnoreRecord(record);
                if (record.getOperationId() != 0) {
-                  log.debugf("Check for thread %d continues from operation %d",
-                        record.getThreadId(), record.getOperationId());
-               }
-            }
-            if (ignoreDeadCheckers) {
-               Object ignored = basicCache.get(ignoredKey(slaveIndex, record.getThreadId()));
-               if (ignored != null && record.getOperationId() <= (Long) ignored) {
-                  log.debugf("Ignoring operations %d - %d for thread %d", record.getOperationId(), ignored, record.getThreadId());
-                  while (record.getOperationId() <= (Long) ignored) {
-                     record.next();
-                  }
+                  log.tracef("Check for thread %d continues from operation %d",
+                             record.getThreadId(), record.getOperationId());
                }
             }
             if (trace) {
                log.tracef("Checking operation %d for thread %d on key %d (%s)",
-                     record.getOperationId(), record.getThreadId(), record.getKeyId(), keyGenerator.generateKey(record.getKeyId()));
+                          record.getOperationId(), record.getThreadId(), record.getKeyId(), keyGenerator.generateKey(record.getKeyId()));
             }
             boolean notification = record.hasNotification(record.getOperationId());
             Object value = findValue(record);
             boolean contains = containsOperation(value, record);
+            // TODO move to PrivateLogLogic?
+            // If check is unsuccessful, verify whether complementary key (k') containing current record operation does exist (every 10 unsuccessful
```

Comment:
Well, the last sentence is probably a bit misleading, as this was might have been fixed as a part of https://github.com/mcimbora/radargun/commit/ad5ea131fac80b3d130fc87372c343e5e3ce1a1b, skipping timeout check when using ignoreDeadCheckers. Nevertheless this change is valid with other scenarios as well - if both key (e.g. [1,2]) and its complement ([1,2,3]) exist in the cache, we always return the first value (1,2), ignoring the fact value 3 is already present in the cache. In this case I believe checker should proceed and check operation 3, refreshing the timestamps.